src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,158 @@
+import { Observable } from "rxjs/Observable";
+import compareVersions from "compare-versions";
+import { request, RequestResponse } from "@dcos/http-service";
+import { BehaviorSubject } from "rxjs";
+
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+export const UpdateStreamType = Symbol("UpdateStreamType");
+const CHECK_DELAY = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION_SETTING = "dcosUIVersion";
+const DISMISSED_VERSION = "dismissedVersion";
+const LAST_TIME_CHECK = "lastTimeCheck";
+
+interface ContentTypeObject {
+  action: string;
+  actionType: string;
+  entity: string;
+  version: string;
+}
+
+interface DCOSUserSettings {
+  dcosUIVersion: DCOSUIVersion;
+}
+
+interface DCOSUIVersion {
+  dismissedVersion: string | number | null;
```

Comment:
Because `getFromLocalStorage` need to return a string or a number, since I use it for both the version and the last time check. I added a potential string type change here - https://github.com/dcos/dcos-ui/pull/3398/files/b3ddf954ad5b7083f5b0d7a79652ad5b4632de89#diff-f53b1374c8ef3fe10aefc772e8eef6ccR146 and I couldn't break it while testing it.